qadence/backends/pulser/pulses.py

```diff
@@ -214,3 +221,33 @@ def digital_rot_pulse(
     )
 
     return Pulse.ConstantDetuning(amplitude=amplitude_wf, detuning=0, phase=phase)
+
+
+def digital_z_rot_pulse(
+    angle: TVar | float, phase: float, channel: Channel, config: Configuration | None = None
+) -> Pulse:
+    if config is None:
+        max_det = channel.max_abs_detuning
+    else:
+        max_det = config.detuning if config.detuning is not None else channel.max_abs_detuning
+
+    # get pulse duration in ns
+    duration = 1000 * abs(angle) / max_det
+
+    # create amplitude waveform
+    amp_wf = SquareWaveform.from_duration(
+        duration=duration,  # type: ignore
+        max_amp=0.0,  # type: ignore[arg-type]
+        duration_steps=channel.clock_period,  # type: ignore[attr-defined]
+        min_duration=channel.min_duration,
+    )
```

Comment:
Is the amplitude waveform needed by the `Pulse` constructor or you can just omit it since it has zero drive?

Reply:
Yes it is needed - it's the same situation as with analog blocks.